core/src/test/java/software/amazon/awssdk/core/async/FileAsyncRequestPublisherTckTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.async;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.tck.TestEnvironment;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public class FileAsyncRequestPublisherTckTest extends org.reactivestreams.tck.PublisherVerification<ByteBuffer> {
+
+    final File example = File.createTempFile("example", ".tmp");
+    final File fileDoesNotExist = new File(example.getPath() + "-does-not-exist");
+
+    public FileAsyncRequestPublisherTckTest() throws IOException {
+        super(new TestEnvironment());
+    }
+
+    @Override
+    public Publisher<ByteBuffer> createPublisher(long l) {
+        return AsyncRequestProvider.fromFile(example.toPath());
+    }
+
+    @Override
+    public Publisher<ByteBuffer> createFailedPublisher() {
+        return AsyncRequestProvider.fromFile(fileDoesNotExist.toPath());
+    }
+}
```

Comment:
simple usage of the TCK. If you want me to submit a complete PR for these things, I'd suggest we use JimFS here, so we can avoid real IO in the testsuite. Would it be ok to include it in the test dependencies? https://github.com/google/jimfs

We're using jimfs for testing such Publishers etc in Akka and it's been pretty reliable.

Reply:
Shouldn't have any problems using this from a licensing perspective (it's Apache 2.0) - and I'd be OK adding as long as it's only in the test scope.

Reply:
Excellent, thanks for confirming